realm/realm-library/src/main/cpp/io_realm_internal_OsObject.cpp

```diff
@@ -165,7 +165,11 @@ static inline size_t do_create_row(jlong shared_realm_ptr, jlong table_ptr)
     auto& shared_realm = *(reinterpret_cast<SharedRealm*>(shared_realm_ptr));
     auto& table = *(reinterpret_cast<realm::Table*>(table_ptr));
     shared_realm->verify_in_write();
+#if REALM_ENABLE_SYNC
+    return sync::create_object(shared_realm->read_group(), table);
```

Comment:
Yes, OS did. But those similar logic exist in `object_accessor.hpp` which we are not ready to use right now. Mostly because of the OS schema. See https://github.com/realm/realm-object-store/pull/434/files#diff-99a0771b4ad4157fdbcf45dfdc8f1e24R184